account-history/unsettled-amounts/demo.js

```diff
@@ -0,0 +1,242 @@
+                }
+            }
+        ).then(function (response) {
+            if (response.ok) {
+                response.json().then(function (responseJson) {
+                    if (responseJson.Exchanges.length > 0) {
+                        resetDropdownOptions("ExchangeId")
+                        setExchangeDropdown(responseJson)
+                    }
+                    console.log(parseResponse(responseJson, "Exchanges", "/hist/v1/unsettledamounts/exchanges", parameters) + JSON.stringify(responseJson, null, 2))
+                });
+            } else {
+                demo.processError(response);
+            }
+        }).catch(function (error) {
+            console.error(error);
+        });
+    }
+
+    /**
+     * Get the unsettled amounts for a specific client, for a specific exchange
+     * @return {void}
+     */
+    function getUnsettledAmountsForExchange() {
+        let parameters = "?ClientKey=" + encodeURIComponent(demo.user.clientKey)
+        AmountTypeSource = read("AmountTypeSource")
+        if (read("ExchangeId") === "-") {
+            console.log("You must select an ExchangeId. \nIf the dropdown is empty, execute the 'Get amounts by exchange' first")
+            return
+        }
+        if (AmountTypeSource !== "All") parameters += "&AmountTypeSource=" + AmountTypeSource
+        fetch(
+            demo.apiUrl + "/hist/v1/unsettledamounts/exchanges/" + read("ExchangeId") + parameters,
+            {
+                "method": "GET",
+                "headers": {
+                    "Authorization": "Bearer " + document.getElementById("idBearerToken").value
+                }
+            }
+        ).then(function (response) {
+            if (response.ok) {
+                response.json().then(function (responseJson) {
+                    console.log(parseResponse(responseJson, "Currencies", "/hist/v1/unsettledamounts/exchanges/" + read("ExchangeId"), parameters) + JSON.stringify(responseJson, null, 2))
+                });
+            } else {
+                demo.processError(response);
+            }
+        }).catch(function (error) {
+            console.error(error);
+        });
+    }
+    demo.setupEvents([
+        { "evt": "click", "elmId": "idBtnGetUnsettledAmountsByCurrency", "func": function () { getUnsettledAmountsByCurrency("Currencies") }, "funcsToDisplay": [getUnsettledAmountsByCurrency], "functionToRun": getUnsettledAmountsByCurrency },
+        { "evt": "click", "elmId": "idBtnGetUnsettledAmountsByAmountType", "func": function () { getUnsettledAmountsByCurrency("AmountTypes") }, "funcsToDisplay": [getUnsettledAmountsByCurrency], "functionToRun": getUnsettledAmountsByCurrency },
+        { "evt": "click", "elmId": "idBtnGetUnsettledAmountsByInstruments", "func": function () { getUnsettledAmountsByInstruments() }, "funcsToDisplay": [getUnsettledAmountsByInstruments], "functionToRun": getUnsettledAmountsByInstruments },
```

Comment:
I don't understand this. Will this be picked up by the boilerplate.js?

Reply:
you mean the "func" key? ? It is just a wrapper function, so I can call some endpoints with input params. Or do you mean something else?

Reply:
Think i removed the confusing part--let me know